client/src/components/RelatedProducts/Cards/Comparison.jsx

```diff
@@ -1,33 +1,48 @@
-import React from 'react';
+import { React } from 'react';
 import styled from 'styled-components';
-import sampleFeatures from '../sampledata/sampleFeatures';
-import Features from './Features';
+import { useFeature } from './FeatureContext';
 
 // should get an array with two arrays of features from each item
 function Comparison() {
-  const item1 = sampleFeatures[0];
-  const item2 = sampleFeatures[1];
+  const twoProductsArray = useFeature();
 
-  return (
-    <Testing>
-      <div>
+  if (twoProductsArray) {
+    const item1 = twoProductsArray.data[0].data;
+    const item2 = twoProductsArray.data[1].data;
 
-        <Features item1={item1} item2={item2} />
+    const FeatArray = item1.features.concat(item2.features);
 
-      </div>
-    </Testing>
+    const item1Array = [];
+    item1.features.map((item) => item1Array.push(item));
 
-  );
-}
+    const item2Array = [];
+    item2.features.map((item) => item2Array.push(item));
+
+    return (
+      <table>
+        <thead>
+          <tr>
+            <th>{item1.name}</th>
+            <th> </th>
+            <th>{item2.name}</th>
+          </tr>
+        </thead>
+        <tbody>
+          {FeatArray.map((item, key) => (
+            <tr key={key}>
+
+              <td>{item1Array.map((item1) => (item1.feature === item.feature ? 'X' : ' '))}</td>
 
-const Testing = styled.div`
-  border-style: solid;
-  height: max-content;
-  width: max-content;
-`;
+              <td>{item.value ? `${item.feature}: ${item.value}` : `${item.feature}`}</td>
+
+              <td>{item2Array.map((item2) => (item2.feature === item.feature ? 'X' : ' '))}</td>
```

Comment:
It doesn't really matter it's limited to 2 products. I'd generally recommend to try to make the code DRY (i.e., 38 and 34 seems very similar, could potentially make number of products as variable to make it flexible say business requirement changed to four or five ).